internal/cli/universal_login_customize.go

```diff
@@ -18,9 +46,195 @@ func customizeUniversalLoginCmd(_ *cli) *cobra.Command {
 		Example: `  auth0 universal-login customize
   auth0 ul customize`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return fmt.Errorf("not yet implemented")
+			var universalLoginBrandingData *universalLoginBrandingData
+
+			if err := ansi.Spinner("Fetching Universal Login branding data", func() (err error) {
+				universalLoginBrandingData, err = fetchUniversalLoginBrandingData(cmd.Context(), cli.api, cli.tenant)
+				return err
+			}); err != nil {
+				return err
+			}
+
+			cli.renderer.JSONResult(universalLoginBrandingData)
+
+			return nil
 		},
 	}
 
 	return cmd
 }
+
+func fetchUniversalLoginBrandingData(
+	ctx context.Context,
+	api *auth0.API,
+	tenantDomain string,
+) (*universalLoginBrandingData, error) {
+	group, ctx := errgroup.WithContext(ctx)
+
+	group.Go(func() (err error) {
+		return ensureCustomDomainIsEnabled(ctx, api)
+	})
+
+	var authenticationProfile *management.Prompt
+	group.Go(func() (err error) {
+		authenticationProfile, err = api.Prompt.Read(ctx)
+		return err
+	})
+
+	var brandingSettings *management.Branding
+	group.Go(func() (err error) {
+		brandingSettings = fetchBrandingSettingsOrUseDefaults(ctx, api)
+		return nil
+	})
+
+	var currentTemplate *management.BrandingUniversalLogin
+	group.Go(func() (err error) {
+		currentTemplate = fetchBrandingTemplateOrUseEmpty(ctx, api)
+		return nil
+	})
+
+	var currentTheme *management.BrandingTheme
+	group.Go(func() (err error) {
+		currentTheme = fetchBrandingThemeOrUseDefault(ctx, api)
+		return nil
+	})
+
+	var tenant *management.Tenant
+	var prompt *promptData
+	group.Go(func() (err error) {
+		tenant, err = api.Tenant.Read(ctx)
+		if err != nil {
+			return err
+		}
+
+		defaultPrompt := "login"
```

Comment:
Seeing this makes me wonder if we make the system consistent by setting defaults only on the CLI-side or UI-side. I suppose in theory that defaults could be useful for both but we'll probably want to coordinate those. Thinking it through, we could make the UI simpler by just having most initial state being inherited by the CLI rather than needing to reconcile.

Not something we need to solve right now, mostly thinking out loud.